cmd/skaffold/app/cmd/parse_config.go

```diff
@@ -108,11 +108,18 @@ func getConfigs(cfgOpts configOpts, opts config.SkaffoldOptions, r *record) ([]*
 func processEachConfig(config *latest.SkaffoldConfig, cfgOpts configOpts, opts config.SkaffoldOptions, r *record, index int) ([]*latest.SkaffoldConfig, error) {
 	// check that the same config name isn't repeated in multiple files.
 	if config.Metadata.Name != "" {
-		prevConfig, found := r.configNameToFile[config.Metadata.Name]
-		if found && prevConfig != cfgOpts.file {
-			return nil, fmt.Errorf("skaffold config named %q found in multiple files: %q and %q", config.Metadata.Name, prevConfig, cfgOpts.file)
+		prev, found := r.configNameToFile[config.Metadata.Name]
+		if found {
+			sl := strings.SplitN(prev, "@", 2) // map value is always formatted as `file_name@config_index`
```

Comment:
This overloading of `configNameToFile` seems error-prone (`@` is a valid character in file names).  It also seems unnecessary since it's entirely our code: we could just have the map take `string` &rarr; `struct { name, index }`.  But more to the point, we don't seem to use this index anywhere else, so couldn't this search-for-duplicates just use a local variable to maintain its state?

And maybe this search-for-duplicates should be done separately?

Reply:
don't know what I was doing here 🤦🏽‍♂️. Removed.

Reply:
We can't effectively maintain all info in a local variable since we also want to print out the file names that cause the conflict.